zipkin-storage/influxdb/src/main/java/zipkin/storage/influxdb/InfluxDBSpanStore.java

```diff
@@ -0,0 +1,114 @@
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.storage.influxdb;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import zipkin2.Call;
+import zipkin2.DependencyLink;
+import zipkin2.Span;
+import zipkin2.storage.QueryRequest;
+import zipkin2.storage.SpanStore;
+
+final class InfluxDBSpanStore implements SpanStore {
+
+  final boolean strictTraceId;
+  final InfluxDBStorage storage;
+
+  InfluxDBSpanStore(InfluxDBStorage influxDB) {
+    this.strictTraceId = influxDB.strictTraceId();
+    this.storage = influxDB;
+  }
+
+  @Override public Call<List<List<Span>>> getTraces(QueryRequest request) {
+    // TODO: spanName is optional
+    String q = String.format("SELECT * FROM \"%s\" WHERE \"service_name\" = '%s' AND \"name\" = '%s' AND time < %dms AND time > %dms",
+      this.storage.measurement(),
+      request.serviceName(),
+      request.spanName(),
+      request.endTs(),
+      request.endTs() - request.lookback()
+    );
+    StringBuilder result = new StringBuilder();
+
+    for (Iterator<Map.Entry<String, String>> i = request.annotationQuery().entrySet().iterator(); i.hasNext(); ) {
+      Map.Entry<String, String> next = i.next();
+      String k = next.getKey();
+      String v = next.getValue();
+      if (v.isEmpty()) {
+        result.append(String.format("\"annotation_key\" = '%s'", k));
+      } else {
+        result.append(String.format("(\"annotation_key\" = '%s' AND \"annotation_value\" = '%s'", k, v));
+      }
+      if (i.hasNext())
+        result.append(" and ");
+    }
+    if (result.length() > 0) {
+      q += result.toString();
+    }
+
+    q += String.format(" AND \"duration\" >= %d ", request.minDuration());
+    q += String.format(" AND \"duration\" <= %d ", request.maxDuration());
+    q += " GROUP BY \"trace_id\" ";
+    q += String.format(" SLIMIT %d ORDER BY time DESC", request.limit());
+
+    Query query = new Query(q, this.storage.database());
+    QueryResult qresult = this.storage.get().query(query);
+
+    throw new UnsupportedOperationException();
+  }
+
+  @Override public Call<List<Span>> getTrace(String traceId) {
+    // make sure we have a 16 or 32 character trace ID
+    traceId = Span.normalizeTraceId(traceId);
+
+    // Unless we are strict, truncate the trace ID to 64bit (encoded as 16 characters)
+    if (!strictTraceId && traceId.length() == 32) traceId = traceId.substring(16);
```

Comment:
ps this part only works if you can match on a substring of the trace ID. For example, in cassandra, we store the trace_id as either whatever was passed or right-16 characters when strictId=false. when strictId = false, we store the other part in a field trace_id_high. This field is ignored and never set unless strictId=false

In elasticsearch, we can change indexing to actually store traceID twice when strictId=false (via tokenizing). How to handle this in influxdb will depend on features.